src/main/java/org/fxmisc/flowless/TargetPosition.java

```diff
@@ -30,6 +30,9 @@ public TargetPosition transformByChange(
         if(itemIndex >= pos + removedSize) {
             // change before the target item, just update item index
             return new StartOffStart(itemIndex - removedSize + addedSize, offsetFromStart);
+        } else if (pos <= itemIndex && itemIndex <= pos + removedSize) {
+            // deletion occurred before viewport and partially into it, so update the item index
+            return new StartOffStart(itemIndex - removedSize + addedSize, offsetFromStart);
         } else if(itemIndex >= pos) {
             // target item deleted, show the first inserted at the target offset
             return new StartOffStart(pos, offsetFromStart);
```

Comment:
Mm... true. I misread the code.

However, that means the bug occurs elsewhere because the TargetPosition in this case is
`StartOffStart(0, 0.0)`.